backend/endpoints/rooms.py

```diff
@@ -62,3 +63,77 @@ async def create_room(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="unable to create room",
     )
+
+
+@router.put(
+    "/org/{org_id}/rooms/{room_id}/members/{member_id}",
+    status_code=status.HTTP_200_OK,
+    responses={
+        400: {"detail": "the max number for a Group_DM is 9"},
+        401: {"detail": "member not an admin"},
+        403: {"detail": "DM room or not found"},
+    },
+)
+async def add_to_room(
+    data: AddToRoom,
+    org_id: str,
+    room_id: str,
+    member_id: str,
+    background_tasks: BackgroundTasks,
+):
+    """Adds a new member(s) to a room
+    Args:
+        data: a pydantic schema that defines the request params
+        org_id (str): A unique identifier of an organisation
+        room_id: A unique identifier of the room to be updated
+        member_id: A unique identifier of the member initiating the request
+
+    Returns:
+        HTTP_200_OK: member added
+    Raises:
+        HTTP_400_BAD_REQUEST: the max number for a Group_DM is 9
+        HTTP_401_UNAUTHORIZED: member not in room or not an admin
+        HTTP_403_FORBIDDEN: DM room or not found
+    """
+
+    DB = DataStorage(org_id)
+    new_member = data.dict().get("new_member")
+    room = await get_room(org_id=org_id, room_id=room_id)
+    member = room.get("room_members").get(str(member_id))
+
+    if room is None or room["room_type"] == RoomType.DM:
+        raise HTTPException(status_code=403, detail="DM room or not found")
+
+    if member is None or member["role"] != Role.ADMIN:
+        raise HTTPException(
+            status_code=401, detail="member not in room or not an admin"
+        )
+
+    if room["room_type"] == RoomType.CHANNEL:
+        room["room_members"].update(new_member)
+
+    if room["room_type"] == RoomType.GROUP_DM:
+        for person in list(new_member.keys()):
```

Comment:
Can we avoid this for loop and just do instead
`
room['room_members'].update(new_data) # this will update the new room with the new members
if len(room["room_members"].keys()) >= 9:
                raise HTTPException(
                    detail="the max number for a Group_DM is 9",
                    status_code=400,
                ) # this then the checks the total members of people in the room including those added
`

Reply:
the loop was put in place in case of bulk adding of members to a group dm so that the count will prevent the update when the len(room["room_members"].keys()) is > 9 at a particular iteration

Reply:
oh! you mean update the members list before I check len(room["room_members"].keys()) right?

Reply:
done! work great